src/ontology.rs

```diff
@@ -745,6 +758,52 @@ impl Ontology {
         self.omim_diseases.values()
     }
 
+    /// Returns a vector contains diseases whose names contains the provided
+    /// substring.
+    ///
+    /// If no such substring is present, the vector will be empty.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use hpo::Ontology;
+    /// let ontology = Ontology::from_binary("tests/example.hpo").unwrap();
+    ///
+    /// for result in ontology.diseases_by_name("Cystinosis") {
+    ///     println!("{:?}", result.name());
+    ///  }
+    /// ```
+    pub fn diseases_by_name<'a>(
+        &'a self,
+        symbol: &'a str,
```

Comment:
I suggest to change the variable name from `symbol` to `substring` or `query` to make it more obvious that this is a substring query and not searching for the full name.